contributing/code/bc.rst

```diff
@@ -258,6 +306,91 @@ Remove constant                                     No
+Use another trait                                   Yes
+**Public Properties**
+Add public property                                 Yes
+Remove public property                              No
+Reduce visibility                                   No
+Move to a used trait                                Yes
+**Protected Properties**
+Add protected property                              Yes
+Remove protected property                           No
+Reduce visibility                                   No
+Make public                                         No
+Move to a used trait                                Yes
+**Private Properties**
+Add private property                                Yes
+Remove private property                             No
+Make public or protected                            Yes
+Move to a used trait                                Yes
+**Constructors and destructors**
+Have constructor or destructor                      No
+**Public Methods**
+Add public method                                   Yes
+Remove public method                                No
+Change name                                         No
+Reduce visibility                                   No
+Make final                                          No [6]_
+Move to used trait                                  Yes
+Add argument without a default value                No
+Add argument with a default value                   No
+Remove argument                                     No
+Add default value to an argument                    No
+Remove default value of an argument                 No
+Add type hint to an argument                        No
+Remove type hint of an argument                     No
+Change argument type                                No
+Change return type                                  No
+**Protected Methods**
+Add protected method                                Yes
+Remove protected method                             No
+Change name                                         No
+Reduce visibility                                   No
+Make final                                          No [6]_
+Make public                                         No [8]_
+Move to used trait                                  Yes
+Add argument without a default value                No
+Add argument with a default value                   No
+Remove argument                                     No
+Add default value to an argument                    No
+Remove default value of an argument                 No
+Add type hint to an argument                        No
+Remove type hint of an argument                     No
+Change argument type                                No
+Change return type                                  No
+**Private Methods**
+Add private method                                  Yes
+Remove private method                               No
+Change name                                         No
+Make public or protected                            Yes
```

Comment:
This conflicts with user added functions, right?

Reply:
same answer as above: since adding a new method is allowed, making a private one public or protected should be allowed

Reply:
But if we can can add methods in core whatever their visibility we should not be able to guarantee BC on adding some in user land, I surely miss something.

Reply:
We do not actually - or we would be just blocked for any future improvements.
Didn't ever you add any public or protected method to any clas already?

Reply:
Ok I agree, this is obvious, but I thought that at some point we documented that it was covered.

Maybe we could precise that when using a trait, composition might break when the trait add methods, because as obvious it is, it already caused issues when adding `json` or `file` shortcuts in the core. WDYT?

Reply:
Is there anything specific to traits?

Reply:
No of course, the same goes for extending any class from the core. Maybe we lack a sentence to explicit that it's not possible. Not all new comers have a strong PHP logic helping make that obvious.

Reply:
@HeahDude do you think we need that extended explanation in this PR ... or can be done later in a separate PR? Thanks!

Reply:
Im' not even sure we should at all actually. So definitely not blocking here.